src/modules/launcher/Plugins/Microsoft.Plugin.Program/Main.cs

```diff
@@ -70,37 +71,40 @@ public void Save()
 
         public List<Result> Query(Query query)
         {
-            foreach (var programArgumentParser in _programArgumentParsers)
-            {
-                if (!programArgumentParser.Enabled)
-                {
-                    continue;
-                }
+            var sources = _programArgumentParsers
+                .Where(programArgumentParser => programArgumentParser.Enabled);
 
+            foreach (var programArgumentParser in sources)
+            {
                 if (!programArgumentParser.TryParse(query, out var program, out var programArguments))
                 {
                     continue;
                 }
 
-                var results1 = _win32ProgramRepository.AsParallel()
-                    .Where(p => p.Enabled)
-                    .Select(p => p.Result(program, programArguments, _context.API));
-
-                var results2 = _packageRepository.AsParallel()
-                    .Where(p => p.Enabled)
-                    .Select(p => p.Result(program, programArguments, _context.API));
+                return Query(program, programArguments).ToList();
+            }
 
-                var result = results1.Concat(results2).Where(r => r != null && r.Score > 0);
-                if (result.Any())
-                {
-                    var maxScore = result.Max(x => x.Score);
-                    result = result.Where(x => x.Score > Settings.MinScoreThreshold * maxScore);
-                }
+            return new List<Result>(0);
+        }
 
-                return result.ToList();
+        private IEnumerable<Result> Query(string program, string programArguments)
+        {
+            var result = _win32ProgramRepository
+                .Concat<IProgram>(_packageRepository)
+                .AsParallel()
+                .Where(p => p.Enabled)
+                .Select(p => p.Result(program, programArguments, _context.API))
+                .Where(r => r?.Score > 0)
+                .ToArray();
```

Comment:
[nit] .ToArray() can be removed

Reply:
The ToArray() is needed so that it wont iterate multiple times (for .Max and ,Where). So it is essential here.